classylss/power/transfers.py

```diff
@@ -0,0 +1,241 @@
+import numpy as np
+
+available = ['CLASS', 'EisensteinHu', 'NoWiggleEisensteinHu']
+
+class CLASS(object):
+    """
+    The linear matter transfer function using the CLASS Boltzmann code.
+
+    Parameters
+    ----------
+    cosmo : :class:`Cosmology`
+        the cosmology instance
+    z : float
+        the redshift of the power spectrum
+    """
+    def __init__(self, cosmo, z):
+
+        # make sure we have Pk
+        if not cosmo.has_pk_matter:
+            cosmo = cosmo.clone(output='Pk')
+        self.cosmo = cosmo
+
+        # find the low-k amplitude to normalize to 1 as k->0 at z = 0
+        self._norm = 1.0; self.z = 0
+        self._norm = 1. / self(1e-8)
+
+        # the proper redshift
+        self.z = z
+
+    def __call__(self, k):
+        r"""
+        Return the CLASS linear transfer function at :attr:`z`. This
+        computes the transfer function from the CLASS linear power spectrum
+        using:
+
+        .. math::
+
+            T(k) = \left [ P_L(k) / k^n_s  \right]^{1/2}.
+
+        We normalize the transfer function :math:`T(k)` to unity as
+        :math:`k \rightarrow 0` at :math:`z=0`.
+
+        Parameters
+        ---------
+        k : float, array_like
+            the wavenumbers in units of :math:`h \mathrm{Mpc}^{-1}`
+
+        Returns
+        -------
+        Tk : float, array_like
+            the transfer function evaluated at ``k``, normalized to unity on
+            large scales
+        """
+        k = np.asarray(k)
+        linearP = self.cosmo.get_pklin(k, self.z) / self.cosmo.h**3 # in Mpc^3
+        primordialP = (k*self.cosmo.h)**self.cosmo.n_s # put k into Mpc^{-1}
+        return self._norm * (linearP / primordialP)**0.5
+
+
+class EisensteinHu(object):
```

Comment:
Yes, I was focusing on the existing functionality in classylss, but we should think about what functionality we want and how to best change/generalize this framework. I think you have a better idea what other sort of use cases exist with CLASS...